### ISSUE TITLE
prometheus support?

### ISSUE THREAD

**Issue by djs55** [mirage/metrics#29]
[prometheus](https://prometheus.io) is a very popular time-series data storage and querying system. It goes well with dashboards like [grafana](https://github.com/grafana/grafana), making pretty pictures like this:

![pretty dashboard](https://camo.githubusercontent.com/d010ea19c70677a0bfd8a64fc01d2b0948e1ffc1/687474703a2f2f646f63732e67726166616e612e6f72672f6173736574732f696d672f66656174757265732f64617368626f6172645f6578312e706e67)

We've got an OCaml implementation of the prometheus metrics gathering format, thanks to @talex5: https://github.com/mirage/prometheus which I've had success with integrating with the [ocaml-qcow](https://github.com/mirage/ocaml-qcow) code: https://github.com/mirage/ocaml-qcow/issues/99

It would be nice if we could easily integrate with existing prometheus infrastructure (and not have to duplicate it)

**Comment by hannesm** [comment]
yes, I agree. In #28 I developed an influx reporter, which I use in a grafana setup as well:

![grafana](https://user-images.githubusercontent.com/228456/49167648-16235d00-f336-11e8-9537-886e889047d6.png)

similar to that, but slightly different would be a prometheus reporter. another useful reporter would be SNMP imho.

**Comment by samoht** [comment]
@CraigFe or @icristescu would you be interested to look at this feature request?

**Comment by hannesm** [comment]
@MagnusS was interested as well -- in influx, a metrics source (e.g. gc) is a measurement, and tags contain the static (indexed) information such as unikernel name / application name. it is not clear to me how to map this information to prometheus.

**Comment by undu** [comment]
I might be worth taking a look at the successor of the prometheus format, which seems to be the de-facto format for metrics in the cloud: https://github.com/OpenObservability/OpenMetrics/blob/master/specification/OpenMetrics.md
I would need a new implementation but it shouldn't be hard to verify as conformance tests seem to be available in the repository.
